### ISSUE TITLE
Analyzer 0.33.3 has improper min SDK constraint

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/sdk#35158]
The latest analyzer doesn't import `dart:async` but references `Future`, which is only valid as of  2.1.0-dev.5.0, but the stated min SDK is 2.0.0-dev.48.0.

This results in the following static error for all users between that sdk and 2.1.0-dev.5.0:

```
file:///.../analyzer-0.33.3/lib/src/dart/analysis/results.dart:180:10: Error: Expected 0 type arguments.
  static Future<ResolvedLibraryResult> tmp(LibraryElement library) async {
         ^
file:///.../analyzer-0.33.3/lib/src/dart/analysis/results.dart:198:12: Error: A value of type '#lib1::ResolvedLibraryResultImpl' can't be assigned to a variable of type 'dart.async::FutureOr<invalid-type>'.
Try changing the type of the left hand side, or casting the right hand side to 'dart.async::FutureOr<invalid-type>'.
    return ResolvedLibraryResultImpl(null, libraryPath, library.source.uri,
           ^
file:///...analyzer-0.33.3/lib/src/dart/analysis/results.dart:180:40: Error: Functions marked 'async' must have a return type assignable to 'Future'.
  static Future<ResolvedLibraryResult> tmp(LibraryElement library) async 
```
The best fix would be to publish a +1 release which adds the `dart:async` import, but no minimum SDK change, and then immediately re-release the current version with an additional +1 but change the minimum SDK constraint to 2.1.0-dev.5.0.

**Comment by stereotype441** [comment]
Published analyzer 0.33.3+1 which adds the `dart:async` import, but no minimum SDK change.  Now preparing a +2 release.

**Comment by jakemac53** [comment]
Thanks Paul!

**Comment by stereotype441** [comment]
Published analyzer 0.33.3+2 which is identical to 0.33.3 but with a minimum SDK constraint of 2.1.0-dev.5.0.
